File: service-parties/Entity/Party.cs
using SBMS.Parties.Entity;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SBMS.Parties.Entity
{
    [Table("parties")]
    public class Party
    {
        [Key]
        [Column("party_id")]
        public long Id { get; set; }

        [Column("business_id")]
        public Guid BusinessId { get; set; }

        [Required]
        [Column("party_type")]
        public PartyType Type { get; set; } // Enum: CUSTOMER or SUPPLIER

        [Required]
        [Column("party_name")]
        public string Name { get; set; } = string.Empty;

        [Column("phone_number")]
        public string? PhoneNumber { get; set; }

        [Column("city")]
        public string? City { get; set; }

        [Column("gstin")]
        public string? Gstin { get; set; }

        [Column("current_balance")]
        public decimal CurrentBalance { get; set; } = 0;
        // Logic: Positive (+) = Receivable, Negative (-) = Payable

        [Column("notes")]
        public string? Notes { get; set; }

        [Column("created_at")]
        public DateTime CreatedAt { get; set; }

        [Column("updated_at")]
        public DateTime UpdatedAt { get; set; }
    }
}

File: service-trading/src/main/java/com/sbms/trading_service/entity/Party.java
package com.sbms.trading_service.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.Immutable;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "parties")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Immutable
public class Party {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "party_id")
    private Long id;

    @Column(nullable = false)
    private UUID businessId; 

    @Enumerated(EnumType.STRING)
    @Column(name = "party_type", nullable = false)
    private PartyType partyType;

    @Column(name = "party_name", nullable = false)
    private String name;

    private String phoneNumber;
    
    private String city;
    
    private String gstin;

    @Column(precision = 12, scale = 2)
    private BigDecimal currentBalance;

    @Column(columnDefinition = "TEXT")
    private String notes;

    private LocalDateTime createdAt;
    
    private LocalDateTime updatedAt;
}